src/backend/distributed/utils/node_metadata.c

```diff
@@ -1424,7 +1424,10 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 	{
 		Name nodeClusterName = DatumGetName(nodeCluster);
 		char *nodeClusterString = NameStr(*nodeClusterName);
-		strlcpy(workerNode->nodeCluster, nodeClusterString, NAMEDATALEN);
+		if (nodeClusterString != NULL)
```

Comment:
This is interesting, it seems that we use the right way to read name field (checked with PostgreSQL examples), and nodecluster is filled with 'default' in pg_dist_node. However, when I print the nodecluster with following logs, it prints NULL.

```
	ereport(WARNING, (errcode(ERRCODE_CONFIG_FILE_ERROR),
					errmsg("workerName: %s", TextDatumGetCString(nodeName))));

	ereport(WARNING, (errcode(ERRCODE_CONFIG_FILE_ERROR),
					errmsg("workerName: %s", workerNode->workerName)));

	{
		Name nodeClusterName = DatumGetName(nodeCluster);
		char *nodeClusterString = NameStr(*nodeClusterName);
		if (nodeClusterString != NULL)
		{
			strlcpy(workerNode->nodeCluster, nodeClusterString, NAMEDATALEN);
		}

		ereport(WARNING, (errcode(ERRCODE_CONFIG_FILE_ERROR),
						errmsg("workerCluster: %s", nodeClusterString)));

		ereport(WARNING, (errcode(ERRCODE_CONFIG_FILE_ERROR),
						errmsg("workerCluster: %s", workerNode->nodeCluster)));
	}
```

Results;
```sql
postgres=# create extension citus;
WARNING:  workerName: localhost
WARNING:  workerName: localhost
WARNING:  workerCluster: (null)
WARNING:  workerCluster:
WARNING:  workerName: localhost
WARNING:  workerName: localhost
WARNING:  workerCluster: (null)
WARNING:  workerCluster:
WARNING:  workerName: localhost
WARNING:  workerName: localhost
WARNING:  workerCluster: (null)
WARNING:  workerCluster:
WARNING:  workerName: localhost
WARNING:  workerName: localhost
WARNING:  workerCluster: (null)
WARNING:  workerCluster:
WARNING:  workerName: localhost
WARNING:  workerName: localhost
WARNING:  workerCluster: (null)
WARNING:  workerCluster:
CREATE EXTENSION
```